netrw/rewire/spatial_small_worlds.py

```diff
@@ -0,0 +1,126 @@
+                    if manhattan_dist:
+                        edge_p = [(dists[0] + dists[1] + dists[2]) for dists in edge_p]
+                    else:
+                        edge_p = [(dists[0]**2 + dists[1]**2 + dists[2]**2)**(1/2) for dists in edge_p]
+                    unique_lengths = np.unique(edge_p)
+                randomVal = random.choices(
+                  unique_lengths, weights=(1 / np.power(unique_lengths,(alpha))), k=1)
+                indices = list(np.where(np.array(edge_p) == randomVal)[0])
+                randomList = random.choices(
+                  [non_edge_list[i] for i in indices], k=1)
+                edge_list = list(G.edges())
+                rand_edge = edge_list[random.randint(0,len(edge_list)-1)]
+                if does_remove:
+                    G.remove_edge(rand_edge[0],rand_edge[1])
+                G.add_edge(randomList[0][0],randomList[0][1])
+            elif dimsize == 2:
+                non_edge_list = list(nx.Graph(nx.non_edges(G)).edges())
+                if not is_periodic:
+                    edge_p = [((edge_pair[0][0]-edge_pair[1][0])**2+(edge_pair[0][1]-edge_pair[1][1])**2)**(1/2) for edge_pair in non_edge_list]
+                    unique_lengths = np.unique(edge_p)
+                else:
+                    edge_p = [(abs(edge_pair[0][0]-edge_pair[1][0]),abs(edge_pair[0][1]-edge_pair[1][1])) for edge_pair in non_edge_list]
+                    edge_p = [(dim[1] - dists[0] if dists[0] > (dim[1])/2 else dists[0] , dim[0] - dists[1] if dists[1] > (dim[0])/2 else dists[1]) for dists in edge_p]
+                    if manhattan_dist:
+                        edge_p = [(dists[0] + dists[1]) for dists in edge_p]
+                    else:
+                        edge_p = [(dists[0]**2 + dists[1]**2)**(1/2) for dists in edge_p]
+                    unique_lengths = np.unique(edge_p)
+                randomVal = random.choices(
+                  unique_lengths, weights=(1 / np.power(unique_lengths,(alpha))), k=1)
+                indices = list(np.where(np.array(edge_p) == randomVal)[0])
+                randomList = random.choices(
+                  [non_edge_list[i] for i in indices], k=1)
+                edge_list = list(G.edges())
+                rand_edge = edge_list[random.randint(0,len(edge_list)-1)]
+                if does_remove:
+                    G.remove_edge(rand_edge[0],rand_edge[1])
+                G.add_edge(randomList[0][0],randomList[0][1])
+            if verbose:
+                if does_remove:
+                    removed_edges[t] = [rand_edge[0],rand_edge[1]]
+                added_edges[t] = [randomList[0][0],randomList[0][1]]
+        if verbose:
+            return G, removed_edges, added_edges
+        else:
+            return G
+
+    def full_rewire(self,G,p,dim,alpha,copy_graph=False,is_periodic=True,does_remove=True,manhattan_dist=True,timesteps=-1,directed=False,verbose=False):
```

Comment:
I think the default value for `copy_graph` in the `full_rewire` method is `True`.